### ISSUE TITLE
Build qCC_db failed with Error C2440 C2664...

### ISSUE THREAD

**Issue by aetzy** [CloudCompare/CloudCompare#2095]
Hi,

I am encountering an issue while compiling CloudCompare that has me quite confused for 2 days. I successfully compiled the CCCoreLib library, but during the compilation of qCC_db, I am running into numerous type conversion errors, as shown in the attached screenshot.

My build environment is: **Windows 10 + QT (5.15.2) + MinGW81_64 (I also tried MSVC2019)**.
I also tested the compilation on another computer and encountered the same problem. The environment there is: **Windows 10 + QT (5.12.11) + MinGW73_64**.

I’ve noticed that these issues seem to be related to type conversions, including C++ type conversions and QString conversions. My computer is set up with **C++14**.

I have previously written C++ MFC code, but I am a newcomer to QT. I believe this issue might be related to my environment, but after searching for two days, I still haven't found the problem. Please help me, I would really appreciate it.

![image](https://github.com/user-attachments/assets/c7d405ac-89da-46bd-8809-97bea17fe446)

**Comment by aetzy** [comment]
I tried Cloudcompare Master branch \ 2.13.2 \ 2.12.4 with the same enviroment, same Error information for these branch.

**Comment by aetzy** [comment]
When I try to build in Ubuntu, there is no build errors.  However I really want to build in Windows

**Comment by dgirardeau** [comment]
Are you maybe trying to compile with Qt6 maybe? (only Qt5 is supported for now)

**Comment by aetzy** [comment]
My first computer does have QT6, but the second computer only has QT5, and it can still reproduce the same error.
![image](https://github.com/user-attachments/assets/fcdefebf-b2a9-4f0d-a31f-d795777ce8ac)

**Comment by dgirardeau** [comment]
Oh wait, I didn't pay attention: have you updated all the submodules? It seems there's a mismatch between CloudCompare and CCCoreLib (libs\qCC_db\extern\CCCoreLib). The latest version of CCCoreLib relies on std::string for managing scalar-field names among other things.

**Comment by aetzy** [comment]
![image](https://github.com/user-attachments/assets/44d518ed-f452-43cf-a568-f48d1a60cc82)
Thank you for your response. I believe I am using the latest version of CCCoreLib. I just downloaded and recompiled the updated CCCoreLib, but I still encounter the following errors when compiling qCC_db. If I manually update these conversion codes, more conversion errors will appear.

**Comment by dgirardeau** [comment]
If it's the latest version of CCCoreLib, then you are not using the latest version of CloudCompare ;)

Aren't you using git to pull all the sources? That's the only solution to get the right version of both repositories (via the 'submodule' mechanism)

**Comment by dgirardeau** [comment]
That's a link issue. You should probably perform a clean build.

**Comment by aetzy** [comment]
Thank you very much for your  help @dgirardeau . I successfully compiled the code, and there were two reasons for my previous failure:

1. I used download instead of git pull, which may have caused version mismatches with the submodules. I hadn't realized this before.
2. On the master branch, I did use a clean build and even deleted the folder, but the link issue persisted. Since I have limited experience with QT, I couldn't identify the cause. I have just switched from MinGW to MSVC2019, and the compilation passed successfully.
